src/conn/conn_api.c

```diff
@@ -1822,9 +1822,46 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
               WT_SINGLETHREAD_STRING));
     }
 
+    /*
+     * We own the database home, figure out if we're creating it. There are a few files created when
+     * initializing the database home and we could crash in-between any of them, so there's no
+     * simple test. The last thing we do during initialization is rename a turtle file into place,
+     * and there's never a database home after that point without a turtle file. If the turtle file
+     * doesn't exist, it's a create.
+     */
+    WT_ERR(__wt_turtle_exists(session, &exist));
+    conn->is_new = exist ? 0 : 1;
+
+    /*
+     * If the turtle file exists, the WiredTiger file should exist as well, unless we are salvaging.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "salvage", &cval));
+    is_salvage = cval.val != 0;
+    if (!is_salvage && !conn->is_new) {
+        WT_ERR(__wt_fs_exist(session, WT_WIREDTIGER, &exist));
+        if (!exist) {
+            F_SET(conn, WT_CONN_DATA_CORRUPTION);
+            WT_ERR_MSG(session, WT_TRY_SALVAGE, "WiredTiger version file cannot be found");
+        }
+    }
+
     /* We own the lock file, optionally create the WiredTiger file. */
-    ret = __wt_open(
-      session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR, is_create ? WT_FS_OPEN_CREATE : 0, &fh);
+    ret = __wt_open(session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR,
+      is_create || is_salvage ? WT_FS_OPEN_CREATE : 0, &fh);
+
+    empty = false;
+    if (fh) {
+        WT_TRET(__wt_filesize(session, fh, &size));
```

Comment:
I am not sure about the use of `WT_TRET`. I would be happy to have opinions from the reviewers. Should we just fail if we cannot retrieve the size and then use `WT_ERR`? @jiechenbo

Reply:
I think in this matter, it would make more sense to place this check after the ret checking of `__wt_open`. This can also removed the need to use `WT_TRET` instead of `WT_ERR`

Reply:
Thanks, that makes sense. I have changed this to `WT_ERR` and moved this section after the ret checking of `__wt_open`.